src/sync/rwlock.rs

```diff
@@ -441,38 +345,23 @@ impl<T: ?Sized> Drop for RwLockReadGuard<'_, T> {
         self.inner = None;
 
         let mut state = self.rwlock.state.borrow_mut();
-
         trace!(
             holder = ?state.holder,
-            waiting_readers = ?state.waiting_readers,
-            waiting_writers = ?state.waiting_writers,
+            semaphore = ?self.rwlock.semaphore,
             "releasing Read lock on rwlock {:p}",
             self.rwlock
         );
-
-        match &mut state.holder {
-            RwLockHolder::Read(readers) => {
-                let was_reader = readers.remove(self.me);
-                assert!(was_reader);
-                if readers.is_empty() {
-                    state.holder = RwLockHolder::None;
-                }
-            }
-            _ => panic!("exiting a reader but rwlock is in the wrong state {:?}", state.holder),
-        }
-
-        if ExecutionState::should_stop() {
-            return;
+        let RwLockHolder::Read(readers) = &mut state.holder else {
+            panic!("exiting a reader but rwlock is in the wrong state {:?}", state.holder);
+        };
+        let was_reader = readers.remove(self.me);
```

Comment:
Could avoid introducing the `was_reader` variable here?

Reply:
I think it was there for clarity, but removed now.